explorer/interpreter/type_checker.cpp

```diff
@@ -5,6 +5,7 @@
 #include "explorer/interpreter/type_checker.h"
 
 #include <algorithm>
+#include <iostream>
```

Comment:
```suggestion
```

We should be using LLVM's io instead of the STL io. Is this include needed?

Note, for temporary debugging statements (which I wonder if this is leftover from), it may be faster to do things like `llvm::errs() << ...`